system/modules/core/forms/FormTextArea.php

```diff
@@ -144,7 +144,7 @@ public function __get($strKey)
 				break;
 
 			case 'value':
-				return specialchars(str_replace('\n', "\n", $this->varValue));
+				return str_replace('\n', "\n", $this->varValue);
```

Comment:
You are right, the current behavior has been there since Contao 3.2.17. I have therefore reverted the changes in e603e0f4b567593ec39d53363c53cccd80ae7017.